types/qs/index.d.ts

```diff
@@ -10,6 +10,7 @@
 //                 Hunter Perrin <https://github.com/hperrin>
 //                 Jordan Harband <https://github.com/ljharb>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.0
```

Comment:
TS 3 is the minimum version that supports `unknown`.

We could support older versions with something like:

```typescript
// The value type here is a "poor man's `unknown`". When these types support TypeScript
// 3.0+, we can replace this with `unknown`.
type PoorMansUnknown = {} | null | undefined;
```

Reply:
Unfortunately `{}` doesn't allow indexing on any key, including strings and numbers, let alone symbols and bigints. i believe DT has a way to do multiple versions, so that TS3+ can use `unknown` while TS2 is relegated to something lesser?

Reply:
> Unfortunately `{}` doesn't allow indexing on any key, including strings and numbers, let alone symbols and bigints.

Isn't that the same as `unknown`, i.e. isn't that what we want? With `unknown` and with `PoorMansUnknown`, the following tests will pass:

```ts
declare const result: { [key: string]: unknown };
result.foo; // $ExpectType unknown
result.foo['bar']; // $ExpectError

declare const result2: { [key: string]: {} | null | undefined };
result2.foo; // $ExpectType {} | null | undefined
result2.foo['bar']; // $ExpectError
```

Reply:
> i believe DT has a way to do multiple versions, so that TS3+ can use `unknown` while TS2 is relegated to something lesser?

There are ways to do this but it doesn't work in every case. You can only augment types for older versions—you can't override, so I'm not sure it would help in this case.